### ISSUE TITLE
Add dictionary_subset()

### ISSUE THREAD

**Issue by koheiw** [quanteda/quanteda#2001]
We can slice the top level of dictionary objects with `[]`, but it is difficult do in nested levels. With a dictionary
```r
dict <- dictionary(list("A" = "a", 
                        "B" = list("A" = "a", "B" = "b")))

```
`dictionary_subset()` should work like
```r
dictionary_subset(dict, keys = "A", levels = 1) 
# the same as dict["A"]

dictionary_subset(dict, keys = "A", levels = 1:2)
# dictionary(list("A" = "a", 
#                 "B" = list("A" = "a")))

dictionary_subset(dict, keys = c("A", "B"), levels = 1:2)
# the same as original
```

**Comment by kbenoit** [comment]
Good use case, I agree this is tricky in the absence of a function or operator for this. 

On naming, it's a syntactic deviation to use `dictionary_subset` since all other subset functions slice documents based on a logical argument.

**Comment by koheiw** [comment]
Then, `dictionary_select()`? or just add the arguments to `dictionary(..., keys = NULL, levels = 1:99)`?

**Comment by kbenoit** [comment]
What about overloading the `[` operator with a `levels` argument? Like in `drop` in `[.data.frame`?

**Comment by koheiw** [comment]
I am fine with `x[key, recur = TRUE]` which applies `[key]` to all the sub-levels. `[, levels = 1:99"]` seems to complex.

**Comment by kbenoit** [comment]
Sounds good. It's a pretty specific use case, but I can see how it would be useful for named elements.

Would it work for numeric indexes, and negative indexes too?

**Comment by koheiw** [comment]
No, logical or numeric does not work when nested levels have different number of keys.  Also I want to only apply `x[]` to nested levels, so `recur` is not very accurate. 

We can treat selection/removal of dictionary elements as editing and use `dictionary_edit()` (it will not for open-up dialog then).
